esmvalcore/preprocessor/regrid_schemes.py

```diff
@@ -0,0 +1,118 @@
+"""Regridding schemes."""
+from __future__ import annotations
+
+import logging
+from collections.abc import Callable
+
+from iris.cube import Cube
+
+from esmvalcore.preprocessor._regrid_esmpy import (
+    ESMPyAreaWeighted,
+    ESMPyLinear,
+    ESMPyNearest,
+)
+from esmvalcore.preprocessor._regrid_unstructured import UnstructuredNearest
+
+logger = logging.getLogger(__name__)
+
+
+__all__ = [
+    'ESMPyAreaWeighted',
+    'ESMPyLinear',
+    'ESMPyNearest',
+    'GenericFuncScheme',
+    'UnstructuredNearest',
+]
+
+
+class _GenericRegridder:
+    r"""Generic function regridder.
+
+    This class can be used in :meth:`iris.cube.Cube.regrid`.
+
+    Parameters
+    ----------
+    src_cube:
+        Cube defining the source grid.
+    tgt_cube:
+        Cube defining the target grid.
+    func:
+        Generic regridding function with signature f(src_cube: Cube, grid_cube:
+        Cube, \*\*kwargs) -> Cube.
+    **kwargs:
+        Keyword arguments for the generic regridding function.
+
+    """
+
+    def __init__(
+        self,
+        src_cube: Cube,
+        tgt_cube: Cube,
+        func: Callable,
+        **kwargs,
+    ):
+        """Initialize class instance."""
+        self.src_cube = src_cube
+        self.tgt_cube = tgt_cube
+        self.func = func
+        self.kwargs = kwargs
+
+    def __call__(self, cube: Cube) -> Cube:
+        """Perform regridding.
+
+        Parameters
+        ----------
+        cube:
+            Cube to be regridded.
+
+        Returns
+        -------
+        Cube
+            Regridded cube.
+
+        """
+        return self.func(cube, self.tgt_cube, **self.kwargs)
+
+
+class GenericFuncScheme:
```

Comment:
Is this scheme useful to users of the Python API? My understanding is that generic schemes are useful to define the scheme in the recipe, but what does it add for the Python user? If that is correct, we could make this scheme private.

Reply:
I don't have a strong opinion on this. This class is only used if a regridding function `f(src_cube, grid_cube, **kws) -> Cube` is available (which is not the case for every generic regridding scheme). This can then be turned into an iris-compatible regridder, which might be helpful for users. Your call.